### PR TITLE
Refactor: Merge client and server `Request`

### PR DESCRIPTION
**Pull request by amitksingh1490**
depends on https://github.com/dream11/zio-http/pull/1094

### CODE REVIEW

**Review by gciuloaica — APPROVED** [review]
looks good to me

File: zio-http/src/test/scala/zhttp/internal/HttpGen.scala [hunk 1]
```diff
@@ -13,28 +12,29 @@ import zio.{Chunk, ZIO}
...
       method  <- HttpGen.method
       url     <- HttpGen.url
       headers <- Gen.listOf(HttpGen.header).map(Headers(_))
-    } yield ClientRequest(url, method, headers, HttpData.fromFile(file))
+      version <- Gen.fromIterable(List(Version.Http_1_0, Version.Http_1_1))
+    } yield Request(version, method, url, headers, data = HttpData.fromFile(file))
   }
 
-  def clientRequest[R](
+  def RequestGen[R](
```
Comment (kaushik-barodiya) [inline zio-http/src/test/scala/zhttp/internal/HttpGen.scala]: Rename to `requestGen`

File: zio-http/src/test/scala/zhttp/internal/HttpGen.scala [hunk 2]
```diff
@@ -13,28 +12,29 @@ import zio.{Chunk, ZIO}
...
 object HttpGen {
-  def clientParamsForFileHttpData(): Gen[Random with Sized, ClientRequest] = {
+  def clientParamsForFileHttpData(): Gen[Random with Sized, Request] = {
     for {
       file    <- Gen.fromEffect(ZIO.succeed(new File(getClass.getResource("/TestFile.txt").getPath)))
       method  <- HttpGen.method
       url     <- HttpGen.url
       headers <- Gen.listOf(HttpGen.header).map(Headers(_))
-    } yield ClientRequest(url, method, headers, HttpData.fromFile(file))
+      version <- Gen.fromIterable(List(Version.Http_1_0, Version.Http_1_1))
```
Comment (kaushik-barodiya) [inline zio-http/src/test/scala/zhttp/internal/HttpGen.scala]: Can we create Gen for the version too? It has been used at couple of places

File: zio-http/src/test/scala/zhttp/http/EncodeClientRequestSpec.scala [hunk 3]
```diff
@@ -2,27 +2,27 @@ package zhttp.http
...
-import zhttp.service.{Client, EncodeClientRequest}
+import zhttp.service.EncodeClientRequest
 import zio.random.Random
 import zio.test.Assertion._
 import zio.test._
 
 object EncodeClientRequestSpec extends DefaultRunnableSpec with EncodeClientRequest {
 
-  val anyClientParam: Gen[Random with Sized, Client.ClientRequest] = HttpGen.clientRequest(
+  val anyClientParam: Gen[Random with Sized, Request] = HttpGen.RequestGen(
```
Comment (kaushik-barodiya) [inline zio-http/src/test/scala/zhttp/http/EncodeClientRequestSpec.scala]: requestGen should be used as `HttpGen.requestGen`

**Review by kaushik143 — CHANGES_REQUESTED** [review]

File: zio-http/src/main/scala/zhttp/service/Client.scala [hunk 4]
```diff
@@ -140,15 +139,19 @@ object Client {
...
     for {
       uri <- ZIO.fromEither(URL.fromString(url))
-      res <- request(ClientRequest(uri, method, headers, content, attribute = Attribute(ssl = Some(ssl))))
+      res <- request(
+        Request(Version.Http_1_1, method, uri, headers, data = content),
+        attribute = Attribute(ssl = Some(ssl)),
+      )
     } yield res
 
   def request(
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Client.scala:148]:
```suggestion
  private [zhttp] def request(
```

File: zio-http/src/test/scala/zhttp/http/EncodeClientRequestSpec.scala [hunk 5]
```diff
@@ -2,27 +2,27 @@ package zhttp.http
 
 import io.netty.handler.codec.http.HttpHeaderNames
 import zhttp.internal.HttpGen
-import zhttp.service.{Client, EncodeClientRequest}
+import zhttp.service.EncodeClientRequest
 import zio.random.Random
 import zio.test.Assertion._
 import zio.test._
 
 object EncodeClientRequestSpec extends DefaultRunnableSpec with EncodeClientRequest {
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/http/EncodeClientRequestSpec.scala]:
```suggestion
object EncodeRequestSpec extends DefaultRunnableSpec with EncodeClientRequest {
```

File: zio-http/src/main/scala/zhttp/service/Client.scala [hunk 6]
```diff
@@ -1,37 +1,34 @@
...
 import zio.{Promise, Task, ZIO}
 
-import java.net.{InetAddress, InetSocketAddress, URI}
+import java.net.{InetSocketAddress, URI}
 
 final case class Client[R](rtm: HttpRuntime[R], cf: JChannelFactory[Channel], el: JEventLoopGroup)
     extends HttpMessageCodec {
 
-  def request(request: Client.ClientRequest): Task[Response] =
+  def request(request: Request, attribute: Attribute): Task[Response] =
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Client.scala]:
```suggestion
  private [zhttp] def request(request: Request, attribute: Attribute): Task[Response] =
```

Comment (amitksingh1490) [reply zio-http/src/main/scala/zhttp/service/Client.scala:26]: Will Add a separate Ticket for this

File: zio-http/src/main/scala/zhttp/service/Client.scala (hunk 4 above)
Comment (amitksingh1490) [reply zio-http/src/main/scala/zhttp/service/Client.scala:148]: Will add a separate issue for this

**Review by kaushik143 — APPROVED** [review]

**Pull request merged**